lib/ansible/modules/cloud/azure/azure_rm_aci.py

```diff
@@ -0,0 +1,500 @@
+                                     ports=[])
+
+    results['containers'] = []
+    if aci.containers:
+        for container in aci.containers:
+            results['containers'].append(create_container_dict_from_obj(container))
+
+    return results
+
+
+class AzureRMContainerInstance(AzureRMModuleBase):
+    """Configuration class for an Azure RM container instance resource"""
+
+    def __init__(self):
+        self.module_arg_spec = dict(
+            resource_group=dict(
+                type='str',
+                required=True
+            ),
+            name=dict(
+                type='str',
+                required=True
+            ),
+            os_type=dict(
+                type='str',
+                required=False,
+                default='linux',
+                choices=['linux', 'windows']
+            ),
+            state=dict(
+                type='str',
+                required=False,
+                default='present',
+                choices=['present', 'absent']
+            ),
+            location=dict(
+                type='str',
+                required=False
+            ),
+            ip_address=dict(
+                type='str',
+                required=False,
+                default=None,
+                choices=['public']
+            ),
+            ports=dict(
+                type='list',
+                required=False
+            ),
+            registry_login_server=dict(
+                type='str',
+                required=False,
+                default=None
+            ),
+            registry_username=dict(
+                type='str',
+                required=False,
+                default=None
+            ),
+            registry_password=dict(
+                type='str',
+                required=False,
+                default=None
+            ),
+            containers=dict(
+                type='list',
+                required=True
+            )
+        )
+
+        self.resource_group = None
+        self.name = None
+        self.location = None
+        self.tags = None
+        self.state = None
+        self.ip_address = None
+
+        self.containers = None
+
+        self.results = dict(changed=False, state=dict())
+        self.mgmt_client = None
+
+        super(AzureRMContainerInstance, self).__init__(derived_arg_spec=self.module_arg_spec,
+                                                       supports_check_mode=True,
+                                                       supports_tags=True)
+
+    def exec_module(self, **kwargs):
+        """Main module execution method"""
+
+        for key in list(self.module_arg_spec.keys()) + ['tags']:
+            setattr(self, key, kwargs[key])
+
+        resource_group = None
+        response = None
+        results = dict()
+        to_be_updated = False
+
+        self.mgmt_client = ContainerInstanceManagementClient(self.azure_credentials, self.subscription_id)
```

Comment:
- We should enable multi-cloud support. The default endpoint is pointing to global Azure.
- You can use `get_mgmt_svc_client` from `AzureRMModuleBase`, which will add Ansible user agent automatically.


You can refer to code snippet at [here](https://github.com/ansible/ansible/blob/devel/lib/ansible/module_utils/azure_rm_common.py#L734-L736).